MAINTAINERS.md

```diff
@@ -77,8 +87,73 @@ Make frequent project releases to the community.
+
+> I propose inviting Vacha Shah ([@vachashah](https://github.com/vachashah)) as a co-maintainer on [opensearch-project/OpenSearch](https://github.com/opensearch-project/OpenSearch). 
+> 
+> Vacha has contributed a lot of automation across many repositories.
+> 
+> - https://github.com/opensearch-project/OpenSearch/pull/877 — Added a broken link checker
+> - https://github.com/opensearch-project/OpenSearch/issues/894 — Fixed broken links across the OpenSearch-Project
+> - https://github.com/opensearch-project/opensearch-build/pull/17  — Updated labels across all OpenSearch repos
+> - https://github.com/opensearch-project/opensearch-plugins/pull/70 — Updated plugin documentation
+> 
+> Vacha has worked tenaciously on non-glamorous tasks, because they raise the overall Engineering bar for the entire project. 
+
+### Interest
+
+Upon receiving at least 3 positive maintainer votes, and no vetoes, from other maintainers within a maximum period of 2 weeks, the nominating maintainer asks a potential nominee whether they might be interested in becoming a maintainer on the repository, which typically happens via e-mail obtained via DCO, but may also be a comment to a pull request, following a substantial contribution.
+
+> This is great work, Vacha! Would you be interested in becoming a [co-maintainer](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) on this project? I’d be honored to nominate you.
+
+Individuals accept the nomination by replying, or commenting, for example _"Thank you! I would love to."_
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a public issue in order to make the vote visible to the community. 
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The issue stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted and the nominating maintainer can thus comment on the issue with _"maintainers has been added"_. If a decision cannot be reached for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary.
```

Comment:
One thing that is unclear to me here: 
it is easy to see the +1's after the 2 weeks, but how do we aggregate the opposing votes? is it a positive outcome when there are more +1's than opposing comments?

Reply:
As written, it's not a straight up or down vote by majority.  If even one maintainer does not support a nomination, it's a veto, and the nomination doesn't go through.  If someone who's not a maintainer is not in favor, the maintainers can and should take it into consideration (and it may move them to veto) but by itself it wouldn't block the nomination.

Reply:
I got a bit mixed up here. So there are two phases for becoming a maintainer once an existing maintainer chose to nominate someone:
1) nomination by at least 3 maintainers + 0 vetos (2 weeks duration)
2) public confirmation - community says what they think and maintainers take it into consideration (2 weeks duration)

If I understand correctly the Public Confirmation is a form of "Speak now or forever hold your peace" kinda thing. Which makes sense to me:)

Reply:
Correct. The 2 weeks is just a time limit to account for inactive maintainers. The first part, and any negative community feedback, happens in private because we are talking about actual humans, and if I have a strong reason why someone shouldn't be a maintainer I'm unlikely to want to share it publicly. The public confirmation is to make the process more transparent and to give an opportunity for non-maintainers to provide feedback.

Reply:
So I wonder if two weeks is really necessary. I understand the desire to permit time for everyone to have a voice, but two weeks as the term seems long and arbitrary. I suspect some nominations might have a longer discussion, while others would seem obvious and everyone voting/opining quickly. Instead of a fixed length, why not suggest a minimum amount of time (72 hours?) and no maximum. (e.g. it stops when votes stops or discussion stops). 72 hours might be too fast, but it would at least guarantee that no one snuck a nomination in over the weekend.

Reply:
What's the rush though? two weeks seems like a good amount of time to me. I propose we start with this and then see how it goes - maybe 72 hours will make more sense as we go along.

Reply:
I picked 2 weeks because people tend to take week long vacations and we want to give them an opportunity to respond. But we can definitely change it later.

Reply:
Just adding some past experience on this one. As @ke4qqq mentioned 72 hours is a pretty [common voting period](https://www.apache.org/dev/pmc.html#send-the-board-a-notice-of-the-vote-to-add-someone). It's pretty much become the standard across all voting activities (nominations, smoke testing, publishing releases) on all Apache projects to avoid long tail requests/decisions.

Here's an example of what happens in practice if many folks are on vacation and a vote doesn't get traction:

```
Only XXXXXX replied, which suggests that other PMC members do not think [the nominee] is ready yet.

Let's close this vote, I will start a new one in the future if [ the nominee] is still active.

This vote has failed.
 ```

In this example a follow up email thread was started:

```
Hi,
In the previous vote for a new committer, I didn't miss the vote but indeed I preferred the option to wait more time for this contributor, while I didn't want to block if there were enough positive votes.
What is the appropriate way to do that, to show that we didn't miss the vote, should we vote +0?
```

With many followups that folks had either missed the vote or forgot to vote `+0`.

I prefer this practice instead of waiting 2 weeks of "I slept since then" or email pileups that cause amnesia. 

This isn't a deal breaker to further hold up this PR, but this PR has "only" been open 10 days. If we had this under vote we would have to wait 4 more days before merging; and you can see several folks getting antsy why this is still open. Food for thought.